### PR TITLE
[New Rule] Disabling User Account Control via Registry

### LINKED ISSUE #863: [New Rule] Disabling User Account Control via Registry
## Description
The Windows registry is a database of configuration rules for a computer. This procedure changes a value to bypass User Access Control (UAC) protection. 

Looking in the registry these are the default values of UAC:

```
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System]
"ConsentPromptBehaviorAdmin"=dword:00000005
"ConsentPromptBehaviorUser"=dword:00000003
"EnableInstallerDetection"=dword:00000001
"EnableLUA"=dword:00000001
"EnableSecureUIAPaths"=dword:00000001
"EnableUIADesktopToggle"=dword:00000000
"EnableVirtualization"=dword:00000001
"FilterAdministratorToken"=dword:00000000
"PromptOnSecureDesktop"=dword:00000001
"ValidateAdminCodeSignatures"=dword:00000000
```

And when the slider is moved to “Never notify” the values changed are

```
"ConsentPromptBehaviorAdmin"=dword:00000000
"EnableLUA"=dword:00000000
"PromptOnSecureDesktop"=dword:00000000
```
Take note that `EnableLUA` has been disabled completely. This is an extremely dangerous value to be in and should never be disabled so its strongly recommend to set this settings to be enabled in group policies so it always gets applied if settings are reset/changed by users or by previously removed malware.

## Required Info

- **Eventing Sources:**
`winlogbeat-*`, `logs-endpoint.events.*`

- **Platforms**
`windows`

## Optional Info
- **References:**
https://www.greyhathacker.net/?p=796

## Syntax
```
registry where event.type:"change" and
  registry.path:(
    "HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\System\\EnableLUA",
    "HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\System\\ConsentPromptBehaviorAdmin",
    "HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\System\\PromptOnSecureDesktop"
    ) and
  registry.data.strings:"0"
```

## Example Data
image

### PR DESCRIPTION
**Pull request by bm11100**
## Issues
resolves #863 

## Summary
User Account Control (UAC) helps prevent malware from damaging a PC and helps organizations deploy a better-managed desktop. With UAC, apps and tasks always run in the security context of a non-administrator account, unless an administrator specifically authorizes administrator-level access to the system. This rule identifies registry value changes to bypass User Access Control (UAC) protection.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/privilege_escalation_disable_uac_registry.toml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+maturity = "production"
+updated_date = "2021/01/20"
+
+[rule]
+author = ["Elastic"]
+description = """
+User Account Control (UAC) helps prevent malware from damaging a PC and helps organizations deploy a better-managed
+desktop. With UAC, apps and tasks always run in the security context of a non-administrator account, unless an
+administrator specifically authorizes administrator-level access to the system. This rule identifies registry value
+changes to bypass User Access Control (UAC) protection.
```
Comment (threat-punter) [inline rules/windows/privilege_escalation_disable_uac_registry.toml:12]:
```suggestion
User Account Control (UAC) can help mitigate the impact of malware on Windows hosts. With UAC, apps and tasks always
run in the security context of a non-administrator account, unless an administrator specifically authorizes
administrator-level access to the system. This rule identifies registry value changes to bypass User Access Control
(UAC) protection.
```

File: rules/windows/privilege_escalation_disable_uac_registry.toml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Disabling User Account Control via Registry"
```
Comment (threat-punter) [inline rules/windows/privilege_escalation_disable_uac_registry.toml:18]:
```suggestion
name = "Disabling User Account Control via Registry Modification"
```

**Review by threat-punter — APPROVED** [review]

**Review by devonakerr — APPROVED** [review]

File: rules/windows/privilege_escalation_disable_uac_registry.toml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+query = '''
+registry where event.type : "change" and
+  registry.path :
+    (
+      "HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\System\\EnableLUA",
+      "HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\System\\ConsentPromptBehaviorAdmin",
+      "HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\System\\PromptOnSecureDesktop"
+    ) and
+  registry.data.strings : "0"
```
Comment (brokensound77) [inline rules/windows/privilege_escalation_disable_uac_registry.toml:38]:
```suggestion
registry where event.type == "change" and
  registry.path :
    (
      "HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\System\\EnableLUA",
      "HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\System\\ConsentPromptBehaviorAdmin",
      "HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\System\\PromptOnSecureDesktop"
    ) and
  registry.data.strings : "0"
```

**Review by brokensound77 — APPROVED** [review]
small tweak and :shipit:

File: rules/windows/privilege_escalation_disable_uac_registry.toml (hunk 3 above)
Comment (bm11100) [reply rules/windows/privilege_escalation_disable_uac_registry.toml:38]: updated

**Pull request merged**